test/integration/nodejs/nodejs_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2022-2024 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package nodejs
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/relabel"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+
+	"github.com/parca-dev/parca-agent/pkg/agent"
+	"github.com/parca-dev/parca-agent/pkg/logger"
+	"github.com/parca-dev/parca-agent/pkg/objectfile"
+	"github.com/parca-dev/parca-agent/pkg/profiler/cpu"
+	"github.com/parca-dev/parca-agent/test/integration"
+)
+
+func TestNodeJS(t *testing.T) {
+	ok, _, err := agent.PreflightChecks(false, false, false)
+	require.Truef(t, ok, "preflight checks failed: %v", err)
+	if err != nil {
+		t.Logf("preflight checks passed but with errors: %v", err)
+	}
+
+	tests := []struct {
+		images  map[string]string
+		program string
+		want    []string
+		wantErr bool
+	}{
+		{
+			images: map[string]string{
+				"17": "17-alpine",
```

Comment:
We definitely need more versions, and we should test images other than `alpine`. We have a known issue for `musl` unwinding so this would potentially fail in any case.

Reply:
Could you please list a few recommended images here? I will try add and test them.

Reply:
18, 19, 20, 21. Just without `-alpine` for now. We can add the alpine variant after we fixed the `musl` issue.

Reply:
ok

Reply:
I have tested the node js 19, it can't work with current config.  I think it's not supported based on https://www.polarsignals.com/docs/nodejs.

Running with node 19 will get us a `perf-map is empty ` error

Reply:
Yeah, let's omit it for now.